File: src/main/java/com/board/growtime/repository/UserRepository.java
package com.board.growtime.repository;

import com.board.growtime.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    
    // GitHub ID로 사용자 찾기
    Optional<User> findByGithubId(String githubId);
    
    // GitHub 로그인으로 사용자 찾기
    Optional<User> findByLogin(String login);
    
    // GitHub ID로 사용자 존재 여부 확인
    boolean existsByGithubId(String githubId);
    
    // GitHub 로그인으로 사용자 존재 여부 확인
    boolean existsByLogin(String login);
} 

File: src/main/kotlin/com/board/growtime/user/UserRepository.kt
package com.board.growtime.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {
    
    // GitHub ID로 사용자 찾기
    fun findByGithubId(githubId: String): Optional<User>
    
    // GitHub 로그인으로 사용자 찾기
    fun findByLogin(login: String): Optional<User>
    
    // GitHub ID로 사용자 존재 여부 확인
    fun existsByGithubId(githubId: String): Boolean
} 